src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getWorkspaceState } from '../../../../core/server/utils';
+import {
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsCheckConflictsObject,
+  OpenSearchDashboardsRequest,
+  SavedObjectsFindOptions,
+} from '../../../../core/server';
+
+type WorkspaceOptions = Pick<SavedObjectsBaseOptions, 'workspaces'> | undefined;
+
+export class WorkspaceIdConsumerWrapper {
+  private formatWorkspaceIdParams<T extends WorkspaceOptions>(
+    request: OpenSearchDashboardsRequest,
+    options?: T
+  ): T {
+    const { workspaces, ...others } = options || {};
+    const workspaceState = getWorkspaceState(request);
+    const workspaceIdParsedFromRequest = workspaceState?.requestWorkspaceId;
+    const workspaceIdsInUserOptions = options?.workspaces;
+    let finalWorkspaces: string[] = [];
+    if (options?.hasOwnProperty('workspaces')) {
+      finalWorkspaces = workspaceIdsInUserOptions || [];
+    } else if (workspaceIdParsedFromRequest) {
+      finalWorkspaces = [workspaceIdParsedFromRequest];
```

Comment:
what is the difference of workspace inside options.workspaces vs get from request? do they co-exist?

Reply:
There are cases that they may co-exist. Using data source for example, in 2.14 all the data source will be created globally(which means no workspaces info will be stored within data source object), and we will set options.workspaces as null to declare that we want to retrieve all the global data sources instead of the data sources within the request.workspaceId.